### PR TITLE
[WFLY-18765] Add Locale parameter while calling toUpperCase and toLowerCase methods

### PR DESCRIPTION
**Pull request by soul2zimate**
Issue: https://issues.redhat.com/browse/WFLY-18765

[WFLY-18765] Add Locale parameter while calling toUpperCase and toLowerCase methods

### CODE REVIEW

**Review by rhusar — APPROVED** [review]
The mod_cluster bit (and others) look good.

Using e.g. turkish locale would break session stickiness if using param since `JSESSIONID` would become

```
[rhusar@ribera ~]$ java ... -Duser.language=tr -Duser.country=TR -Duser.variant=TR Scratch
jsessıonıd
```

**Comment by soul2zimate** [comment]
Hi @bstansberry , May I ask you review and approval on this ? I'd like to backport this to downstream.

File: jdr/src/main/java/org/jboss/as/jdr/util/JdrZipFile.java [hunk 1]
```diff
@@ -151,7 +152,7 @@ public void addAsString(InputStream stream, String path) throws Exception {
 
     public String getProductDirName() {
         if(this.productDirName == null)
-            this.productDirName = String.format("%s-%s", this.env.getProductName().replace(" ", "_").toLowerCase(), this.env.getProductVersion().split("\\.")[0]);
+            this.productDirName = String.format("%s-%s", this.env.getProductName().replace(" ", "_").toLowerCase(Locale.getDefault()), this.env.getProductVersion().split("\\.")[0]);
```
Comment (bstansberry) [inline jdr/src/main/java/org/jboss/as/jdr/util/JdrZipFile.java:155]: For WF and EAP, Locale.ENGLISH is more correct as we're not going to use anything else for this.

But WF is extensible, so it's possible community users are using something else, so +1 to this. It's the same behavior as what we currently have.

Comment (jasondlee) [reply jdr/src/main/java/org/jboss/as/jdr/util/JdrZipFile.java:155]: Why is this the only place that calls (redundantly, as Brian noted) `Locale.getDefault()`?

**Review by bstansberry — APPROVED** [review]
@soul2zimate I've left some comments in case they spark any thoughts from you, but I think this is ok. Thanks!

File: jdr/src/main/java/org/jboss/as/jdr/util/JdrZipFile.java (hunk 1 above)
Comment (bstansberry) [reply jdr/src/main/java/org/jboss/as/jdr/util/JdrZipFile.java:155]: I think explicitly saying 'Locale.getDefault()' is preferable to saying nothing, as it let's us distinguish an intentional choice from the common case of just forgetting to specify a locale.

**Comment by petrberan** [comment]
Hi @bstansberry if those minor changes are not preventing merging the PR and we have donwstream issue for this one, would you mind merging it? If you'd like those changes handled, please let us know

**Comment by soul2zimate** [comment]
Hi @bstansberry,  I don't really have any more to input for this PR. Since this is approved, could you merge it ? Thanks.

**Comment by darranl** [comment]
Large PRs like this are really not great to coordinate PR reviews, this has touched many subsystems and GitHub is not great at indicating if an approval is for a whole PR or a subset so you risk getting stuck waiting for the future reviews.

**Review by bstansberry — APPROVED** [review]

**Review by jasondlee — APPROVED** [review]

**Comment by bstansberry** [comment]
Thanks @soul2zimate, @rhusar and @jasondlee

**Comment by soul2zimate** [comment]
Thank you all for the reviews.

**Pull request merged**